integration-tests/build.rs

```diff
@@ -0,0 +1,247 @@
+            )
+        })?;
+
+        let compiled = CompilerOutput::deserialize(&mut deserializer)
+            .map_err(|err| BuildError::CompilerOutputDeSerError(err.to_string()))?;
+        let compiled_binding = compiled.clone();
+
+        let error_free = !{ compiled.has_error() || compiled.has_warning(WARN) };
+
+        let _ = error_free.then_some(|| ()).ok_or_else(|| {
+            BuildError::CompilerOutputContainsErrors {
+                path: p.to_string(),
+                errors: compiled.errors,
+            }
+        })?;
+
+        let contract: CompactContractRef = compiled_binding.get(p, name).ok_or_else(|| {
+            BuildError::FailedToLoadCompactContractRef("FailedToLoadCompactContractRef".to_string())
+        })?;
+
+        let abi = contract
+            .abi
+            .ok_or_else(|| {
+                BuildError::ErrorLoadingContractAbi("Failed to get contract Abi".to_string())
+            })?
+            .clone();
+
+        let bin = contract
+            .bin
+            .ok_or_else(|| {
+                BuildError::ErrorLoadingContractBin("Failed to get contract Bin".to_string())
+            })?
+            .clone()
+            .into_bytes()
+            .ok_or_else(|| {
+                BuildError::ByteCodeToBytesError("Could not convert bin to bytes".to_string())
+            })?;
+
+        let bin_runtime = contract
+            .bin_runtime
+            .ok_or_else(|| {
+                BuildError::ErrorLoadingContractBinRuntime(
+                    "Failed to get contract bin-runtime".to_string(),
+                )
+            })?
+            .clone()
+            .into_bytes()
+            .ok_or_else(|| {
+                BuildError::ByteCodeToBytesError(
+                    "Could not convert bin-runtime to bytes".to_string(),
+                )
+            })?;
+
+        let compiled_contract = CompiledContract {
+            path: p.to_string(),
+            name: name.to_string(),
+            abi,
+            bin,
+            bin_runtime,
+        };
+
+        // Save CompiledContract object to json files in "contracts" folder
+        let mut path_json = path_sol.clone();
+        path_json.set_extension("json");
+        serde_json::to_writer(
+            &File::create(&path_json).expect("cannot create file"),
+            &compiled_contract,
+        )
+        .map_err(|err| BuildError::JsonSerializatonError(err.to_string()))?;
+    }
+    // Generate contract binding for compiled contracts
+    for entry in glob::glob("./contracts/*.json").unwrap() {
+        match entry {
+            Ok(path) => {
+                let _ = generate_rust_contract_bindings(BINDINGS_DR, &path);
+            }
+            Err(e) => eprintln!("{:#?}", e),
+        }
+    }
+    Ok(())
+}
+
+fn generate_rust_contract_bindings(bindings_dir: &str, file: &Path) -> Result<(), BuildError> {
+    const SLASH: char = '/';
+    const DOT: char = '.';
```

Comment:
Nitpick: The PathBuf type has methods to "split the path by /" with https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.components
And to extract the base of the file https://doc.rust-lang.org/std/path/struct.PathBuf.html#method.file_stem
I think using those methods would simplify a bit this function.

Reply:
thanks for the suggestion. Commit 8c61fd3cd7563827af6b8009d53f0af20b7debf6